File: python/moneysocket/core/message/request/ping.py
# Copyright (c) 2020 Jarret Dyrbye
# Distributed under the MIT software license, see the accompanying
# file LICENSE or http://www.opensource.org/licenses/mit-license.php

from core.message.request.request import MoneysocketRequest


class RequestPing(MoneysocketRequest):
    def __init__(self):
        super().__init__("REQUEST_PING")

    @staticmethod
    def check_valid(msg_txt):
        return super().validate(msg_txt)

File: browser/src/js/moneysocket/core/message/request/ping.js
// Copyright (c) 2020 Jarret Dyrbye
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php


const MoneysocketRequest = require('./request.js').MoneysocketRequest;

let REQUEST_SUBCLASSES = require('./request.js').REQUEST_SUBCLASSES;


class RequestPing extends MoneysocketRequest {
    constructor() {
        super("REQUEST_PING");
    }

    cryptLevel() {
        return "AES";
    }

    static castClass(msg_dict) {
        var c = new RequestPing();
        Object.keys(msg_dict).forEach(key => {
            c[key] = msg_dict[key];
        });
        return c;
    }

    static checkValidMsgDict(msg_dict) {
        return null;
    }
}

REQUEST_SUBCLASSES['REQUEST_PING'] = RequestPing;

exports.RequestPing = RequestPing;
